feat(DRIVERS-2903): use custom aws configuration (#1743)
mongodb/specifications: source/auth/auth.md

application. Alternatively, you can create an AWS profile specifically for your MongoDB credentials and set the
`AWS_PROFILE` environment variable to that profile name."

##### Custom Credential Providers

Drivers that choose to use the AWS SDK to fetch credentials MAY also allow users to provide a custom credential provider
as an option to the `MongoClient`. The interface for the option provided depends on the individual language SDK and
drivers MUST consult AWS SDK documentation to determine that format when implementing. The name of the option MUST be
`AWS_CREDENTIAL_PROVIDER` and be part of the authentication mechanism properties options that can be provided to the
client.

> Drivers that implement this MAY choose to implement the following scenarios when applicable in their language's SDK:

Reviewer: ```suggestion
Drivers MAY expose API for default providers for the following scenarios when applicable in their language's SDK:
```

Suggest clarifying the suggestions are API for default providers.

application. Alternatively, you can create an AWS profile specifically for your MongoDB credentials and set the
`AWS_PROFILE` environment variable to that profile name."

##### Custom Credential Providers

> Drivers that choose you use the AWS SDK to fetch credentials MAY also allow users to provide a custom credential

Reviewer 2: ```suggestion
Drivers that choose to use the AWS SDK to fetch credentials MAY also allow users to provide a custom credential
```

provider as an option to the `MongoClient`. The interface for the option provided depends on the individual language SDK
and drivers MUST consult AWS SDK documentation to determine that format when implementing. The name of the option MUST
be `AWS_CREDENTIAL_PROVIDER` and be part of the authentication mechanism properties options that can be provided to the
client.

> Drivers that implement this MAY choose to implement the following scenarios when applicable in their labguage's SDK:

Reviewer 2: ```suggestion
Drivers that implement this MAY choose to implement the following scenarios when applicable in their language's SDK:
```

        check MUST be performed after SRV record resolution, if applicable. This property is only required for drivers
        that support the [Human Authentication Flow](#human-authentication-flow).

    - AWS_CREDENTIAL_PROVIDER

>         A function or object from the AWS SDK that can be used to return AWS credentials.

Reviewer: Consider similarly allowing the property to be defined on the `MongoClient` (as is done for `OIDC_HUMAN_CALLBACK`):

```suggestion
        A function or object from the AWS SDK that can be used to return AWS credentials. Drivers MAY allow the user to
        specify the callback using a `MongoClient` configuration instead of a mechanism property, depending on what is
        idiomatic for the driver.
```

I expect this may be preferred over specifying in the URI string.
